idaes_examples/notebooks/docs/unit_models/custom_unit_models/creating_unit_model.ipynb

```diff
@@ -0,0 +1,2285 @@
+    "\n",
+    "# Import Pyomo libraries\n",
+    "from pyomo.environ import (\n",
+    "    Param,\n",
+    "    Set,\n",
+    "    Var,\n",
+    "    NonNegativeReals,\n",
+    "    units,\n",
+    "    Expression,\n",
+    "    PositiveReals,\n",
+    ")\n",
+    "\n",
+    "# Import IDAES cores\n",
+    "from idaes.core import (\n",
+    "    declare_process_block_class,\n",
+    "    MaterialFlowBasis,\n",
+    "    PhysicalParameterBlock,\n",
+    "    StateBlockData,\n",
+    "    StateBlock,\n",
+    "    MaterialBalanceType,\n",
+    "    EnergyBalanceType,\n",
+    "    Solute,\n",
+    "    Solvent,\n",
+    "    LiquidPhase,\n",
+    ")\n",
+    "from idaes.core.util.model_statistics import degrees_of_freedom"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "a76f25a0",
+   "metadata": {},
+   "source": [
+    "# 1.2 Physical Parameter Data Block\n",
+    "\n",
+    "A `PhysicalParameterBlock` serves as the central point of reference for all aspects of the property package and needs to define several things about the package. These are summarized below:\n",
+    "\n",
+    "- Units of measurement\n",
+    "- What properties are supported and how they are implemented\n",
+    "- What components and phases are included in the packages\n",
+    "- All the global parameters necessary for calculating properties\n",
+    "- A reference to the associated State Block class, so that construction of the State Block components can be automated from the Physical Parameter Block\n",
+    "\n",
+    "To construct this block, we begin by declaring a process block class using a Python decorator. One can learn more about `declare_process_block_class` [here](https://github.com/IDAES/idaes-pse/blob/eea1209077b75f7d940d8958362e69d4650c079d/idaes/core/base/process_block.py#L173). After constructing the process block, we define a build function which contains all the components that the property package would have. `super` function here is used to give access to methods and properties of a parent or sibling class and since this is used on the class `PhysicalParameterData` class, build has access to all the parent and sibling class methods. \n",
+    "\n",
+    "The `PhysicalParameterBlock` then refers to the `state block`, in this case `OrgPhaseStateBlock` (which will be declared later), so that we can build a state block instance by only knowing the `PhysicalParameterBlock` we wish to use. Then we move on to list the number of phases in this property package. Then we assign the variable to the phase which follows a naming convention. Like here since the solvent is in the Organic phase, we will assign the Phase as OrganicPhase and the variable will be named Org as per the naming convention. The details of naming conventions can be found [here](https://github.com/IDAES/idaes-pse/blob/main/docs/explanations/conventions.rst). We will be following the same convention throughout the example. \n",
```

Comment:
```suggestion
    "The `PhysicalParameterBlock` then refers to the `state block`, in this case `OrgPhaseStateBlock` (which will be declared later), so that we can build a state block instance by only knowing the `PhysicalParameterBlock` we wish to use. Then we list the number of phases in this property package. Then we assign the variable to the phase which follows a naming convention. Like here since the solvent is in the Organic phase, we will assign the Phase as OrganicPhase and the variable will be named Org as per the naming convention. The details of naming conventions can be found [here](https://github.com/IDAES/idaes-pse/blob/main/docs/explanations/conventions.rst). We will be following the same convention throughout the example. \n",
```